### ISSUE TITLE
iOS 8 AFNetworking NSURLSessionDataTask Stubs Not Being Called

### ISSUE THREAD

**Issue by Megatron1000** [AliSoftware/OHHTTPStubs#104]
I'm using OHHTTPStubs 4.0.2 with AFNetworking 2.5.3 and onStubActivation is never getting called from my network requests. 

I also have Google Analytics installed and if I allow this to make network requests I can stub these using OHHTTPStubs, which leads me to believe there's something incompatible about my own network requests and perhaps it's that I'm using NSURLSessionDataTask types of requests. 

```
+ (instancetype)sharedClient
{
    static JSShopperSessionManager *client = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        client = [[JSShopperSessionManager alloc] initWithCleanServiceURL];
        client.requestSerializer = [AFJSONRequestSerializer serializer];
        client.responseSerializer.acceptableContentTypes = [NSSet setWithObjects:JSContentTypeJSON, JSContentTypeHAL, JSContentTypeProblem, nil];
    });
    return client;
}
```

```
+ (void)createBasketForStoreCode:(NSString *)storeCode withSuccess:(void (^)(JSBasket *basket))success andFailure:(void (^)(NSError *error))failure
{
    NSDictionary *params = @{@"store_id" : storeCode};

    [[JSShopperSessionManager sharedClient] POST:kShoppingServicePath parameters:params success:^(NSURLSessionDataTask *task, id responseObject)
    {
        success(nil);
    } failure:^(NSURLSessionDataTask *task, NSError *error) {
        failure(error);
    }];
}
```

and my unit test looks like this

```
- (void)testCreateBasket
{    
    [OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest *request)
     {
         return [request.URL.absoluteString.lastPathComponent containsString:kShoppingServicePath];
     }
                        withStubResponse:^OHHTTPStubsResponse*(NSURLRequest *request) {
                            NSString* fixture = OHPathForFile(@"EmptyShop.json", self.class);
                            return [OHHTTPStubsResponse responseWithFileAtPath:fixture
                                                                    statusCode:201 headers:@{@"Content-Type":JSContentTypeJSON}];
                        }];
    
    
    XCTestExpectation *expectation = [self expectationWithDescription:@"The request should successfully complete within the specific timeframe."];
    
    [JSShopperSessionManager createBasketForStoreCode:JSWandsworthStoreID
                                          withSuccess: ^(JSBasket *basket) {
                                              XCTFail(@"Mocked network call failed");
                                              [expectation fulfill];
                                          } andFailure: ^(NSError *error) {
                                              XCTFail(@"Mocked network call failed: %@", error);
                                              [expectation fulfill];
                                          }];
    
    
    [self waitForExpectationsWithTimeout:5 handler:nil];
    
}
```

**Comment by AliSoftware** [comment]
Hi,

**How did you integrate `OHHTTPStubs` in your project?**

If you integrated using CocoaPods for example, this should work out of the box.

See my Example project and also my own Unit Tests in the OHHTTPStubs workspace, some of them using `AFNetworking` and `NSURLSession` and working without any issue, see [this one for example](https://github.com/AliSoftware/OHHTTPStubs/blob/master/OHHTTPStubs/UnitTests/Test%20Suites/AFNetworkingTests.m#L177))

If you integrated manually (which I strongly discourage, both because it is more painful to maintain and because it could lead to integration problems if you didn't do it right), you may have done it wrong. For example you may have forgot to add the `-ObjC` flag so that the categories of the library (especially the one that does the OHHTTPStubs injection in NSURLSession classes) is imported as well as its classes.

Se the dedicated page about manual integration in this repo's wiki for more info; but again I strongly discourage manually integrating `OHHTTPStubs` (as well as any third-party library) to your project (that's painful and error-prone), and better encourage well-recognized tools like to manage libs/dependencies, like CocoaPods. See the README for integration instructions.

**Comment by Megatron1000** [comment]
Hi I was installing it via pods.
Don't worry about this issue now. I've managed to figure out what was wrong. Basically I was overriding dataTaskWithRequest:completionHandler: in order to make another asynchronous request for an authorisation token. If I stop doing this it starts to working. The problem was probably because when I override the method I end up returning nil for the data task. I've managed get it to work by stubbing this method on my session manager and passing the method call onto its superclass.
